src/pool-cl/anti-sniping/CLAntiSniping.sol

```diff
@@ -0,0 +1,259 @@
+            firstBlockFeesToken0[poolId][positionKey] =
+                                FullMath.mulDiv(feeGrowthDelta0X128, info.liquidity, FixedPoint128.Q128);
+            firstBlockFeesToken1[poolId][positionKey] =
+                                FullMath.mulDiv(feeGrowthDelta1X128, info.liquidity, FixedPoint128.Q128);
+        }
+
+        delete positionsCreatedInLastBlock[poolId];
+    }
+
+    /// @notice Handles logic after removing liquidity, redistributing first-block fees if applicable.
+    /// @dev Donates first-block accrued fees to the pool if liquidity remains; otherwise, returns them to the sender.
+    function afterRemoveLiquidity(
+        address sender,
+        PoolKey calldata key,
+        ICLPoolManager.ModifyLiquidityParams calldata params,
+        BalanceDelta,
+        BalanceDelta,
+        bytes calldata
+    ) external override returns (bytes4, BalanceDelta) {
+        PoolId poolId = key.toId();
+        bytes32 positionKey = CLPosition.calculatePositionKey(sender, params.tickLower, params.tickUpper, params.salt);
+
+        BalanceDelta hookDelta;
+        if (poolManager.getLiquidity(poolId) != 0) {
+            hookDelta = toBalanceDelta(
+                firstBlockFeesToken0[poolId][positionKey].toInt128(),
+                firstBlockFeesToken1[poolId][positionKey].toInt128()
+            );
+            poolManager.donate(
+                key, firstBlockFeesToken0[poolId][positionKey], firstBlockFeesToken1[poolId][positionKey], new bytes(0)
+            );
+        } else {
+            // If the pool is empty, the fees are not donated and are returned to the sender
+            hookDelta = BalanceDeltaLibrary.ZERO_DELTA;
+        }
+
+        // Cleanup stored data for the position
+        delete positionCreationBlock[poolId][positionKey];
+        delete firstBlockFeesToken0[poolId][positionKey];
+        delete firstBlockFeesToken1[poolId][positionKey];
+
+        return (this.afterRemoveLiquidity.selector, hookDelta);
+    }
+
+    /// @notice Handles logic before adding liquidity, enforcing position creation constraints.
+    /// @dev Records position creation block and ensures the position doesn't already exist or exceed the same block limit.
+    function beforeAddLiquidity(
+        address sender,
+        PoolKey calldata key,
+        ICLPoolManager.ModifyLiquidityParams calldata params,
+        bytes calldata
+    ) external override returns (bytes4) {
+        PoolId poolId = key.toId();
+        collectLastBlockInfo(poolId);
+        bytes32 positionKey = CLPosition.calculatePositionKey(sender, params.tickLower, params.tickUpper, params.salt);
+        LiquidityParams storage liqParams = positionKeyToLiquidityParams[positionKey];
+        liqParams.sender = sender;
+        liqParams.tickLower = params.tickLower;
+        liqParams.tickUpper = params.tickUpper;
+        liqParams.salt = params.salt;
+        positionKeyToLiquidityParams[positionKey] = liqParams;
```

Comment:
liqParams is storage , do not need this line.
` positionKeyToLiquidityParams[positionKey] = liqParams;`

Reply:
ok, removed